packages/nimble-components/src/wafer-map/tests/hover-handler.spec.ts

```diff
@@ -0,0 +1,122 @@
+                value: Float64Array.from([1, 2, 3])
+            }),
+            zoomIdentity,
+            WaferMapOriginLocation.bottomLeft,
+            undefined,
+            getDataManagerMockForHover(
+                { left: 0, right: 0, top: 0, bottom: 0 },
+                getScaleQuantile([1, 11], [1, 2, 3, 4]),
+                getScaleQuantile([1, 11], [1, 2, 3, 4])
+            ) as DataManager
+        ) as WaferMap;
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('will return the expected index when mouse moved in range', () => {
+        const parameters: [WaferMapOriginLocation, HoverDie][] = [
+            [
+                WaferMapOriginLocation.bottomLeft,
+                {
+                    index: 1,
+                    x: 2,
+                    y: 2
+                }
+            ],
+            [
+                WaferMapOriginLocation.topLeft,
+                {
+                    index: 1,
+                    x: 2,
+                    y: 2
+                }
+            ],
+            [
+                WaferMapOriginLocation.bottomRight,
+                {
+                    index: 1,
+                    x: 2,
+                    y: 2
+                }
+            ],
+            [
+                WaferMapOriginLocation.topRight,
+                {
+                    index: 1,
+                    x: 2,
+                    y: 2
+                }
+            ]
+        ];
+        parameters.forEach(value => {
+            waferMock.originLocation = value[0];
+            hoverHandler = new HoverHandler(waferMock);
+            element.addEventListener('mousemove', event => hoverHandler.onMouseMove(event));
+            element.dispatchEvent(
+                new MouseEvent('mousemove', {
+                    clientX: 4,
+                    clientY: 4
+                })
+            );
+            processUpdates();
+            expect(waferMock.hoverDie).toEqual(value[1]);
+        });
+    });
+
+    it('will return undefined when mouse moved out of range', () => {
+        const parameters: [WaferMapOriginLocation, HoverDie | undefined][] = [
+            [WaferMapOriginLocation.bottomLeft, undefined],
+            [WaferMapOriginLocation.topLeft, undefined],
+            [WaferMapOriginLocation.bottomRight, undefined],
+            [WaferMapOriginLocation.topRight, undefined]
+        ];
+        parameters.forEach(value => {
+            waferMock.originLocation = value[0];
+            hoverHandler = new HoverHandler(waferMock);
+            element.addEventListener('mousemove', event => hoverHandler.onMouseMove(event));
+            element.dispatchEvent(
+                new MouseEvent('mousemove', {
+                    clientX: 15,
+                    clientY: 15
+                })
+            );
+            processUpdates();
+            expect(waferMock.hoverDie).toEqual(value[1]);
+        });
+    });
```

Comment:
Lets use the parameterizeSpec patterns instead: https://www.npmjs.com/package/@ni/jasmine-parameterized

Reply:
Should look something like:

```ts
const testCases = [
    {name: WaferMapOriginLocation.bottomLeft, expectedDie: undefined},
    {name: WaferMapOriginLocation.topLeft, expectedDie: undefined},
    {name: WaferMapOriginLocation.bottomRight, expectedDie: undefined},
    {name: WaferMapOriginLocation.topRight, expectedDie: undefined},
] as const;
parameterizeSpec(testCases, (spec, name, value) => {
    spec(`will return undefined when mouse moved out of range from ${name}`, () => {
        waferMock.originLocation = value.name;
        hoverHandler = new HoverHandler(waferMock);
        element.addEventListener('mousemove', event => hoverHandler.onMouseMove(event));
        element.dispatchEvent(
            new MouseEvent('mousemove', {
                clientX: 15,
                clientY: 15
            })
        );
        processUpdates();
        expect(waferMock.hoverDie).toEqual(value.expectedDie);
    });
});
```

Note:
- How the testCases data is marked `as const`
- How the testCases fields are well-named (`name`, `expectedDie`) instead of indexes in an array